packages/ember-principled-forms/tests/dummy/app/components/form-field/component.ts

```diff
@@ -6,42 +6,45 @@ import update from 'ember-object-update';
 import { Maybe } from 'true-myth';
 
 import Field, { validate } from 'principled-forms/field';
-import Number from 'principled-forms/field/number';
+import EmailField from 'principled-forms/field/email';
+import NumberField from 'principled-forms/field/number';
 import Form, { FromModel } from 'principled-forms/form';
+import Validity from 'principled-forms/validity';
 
 // @ts-ignore: Ignore import of compiled template
 import layout from './template';
 
 type User = {
   age: number;
-  name: Maybe<string>;
+  name: string;
+  email: Maybe<string>;
 };
 
 const userForm: FromModel<User> = user => ({
-  age: Number.required({ eager: false }),
-  name: Field.optional()
+  // DISCUSS: Does it actually every make sense for `eager` to be false?
+  // Isn't our need for eagerness solved by having a separate unvalidated state?
+  age: NumberField.required({ eager: false }),
+  name: Field.required(),
+  email: EmailField.optional({ validity: Validity.valid() }), 
 });
 
 type UserForm = Form<User>;
 type FormProp = keyof UserForm;
+// DISCUSS: This should probably be its own type for convenience, e.g. FormValue<T>
```

Comment:
Concur. More to the point, the way this works as well as the way I had it before doesn't actually resolve the types the way I want; we need to do something like this:

```ts
@action
handleChange<K extends FormProp>(property: K, value: FormValue<UserForm[K]>) {
  // ...
}
```